### ISSUE TITLE
Training of my own classification model with very bad results (100 epochs)

### ISSUE THREAD

**Issue by Sgomes14843** [dusty-nv/jetson-inference#671]
Hi, I trained my own dataset following the tutorial "Re-training on the PlantCLEF Dataset". My dataset consists of 10 classes of F1 cars per team. So I trained following the steps of the tutorial, with 100 epochs and batch size of 16. I used resnet50 and then resnet 18. With the onnx model I tested on jetson and the results are bad. The classifier only hits me only one class of 10, otherwise it changes everything, for example saying that a ferrari is a mercedes and etc ...
In the images I attached, he only hit the toro_rosso, and it seems to me that it is the only class that he hits right.
Did the problem have been training with 100 epochs, is more needed?

![116836158_339680790380140_4819763472371178894_n](https://user-images.githubusercontent.com/60934040/89405840-192e3d80-d714-11ea-882a-83153ebcd120.jpg)
![116863875_297779325000874_3046177056885957282_n](https://user-images.githubusercontent.com/60934040/89405841-1a5f6a80-d714-11ea-906a-c2c91e910c62.jpg)
![116872105_1163397294032672_1276804440622163168_n(1)](https://user-images.githubusercontent.com/60934040/89405843-1af80100-d714-11ea-85b1-40c3dbb4c72b.jpg)

**Comment by dusty-nv** [comment]
Hi @Sgomes14843 , what was the accuracy of the model in PyTorch after epoch 100?   How many images per class were in your training dataset?

Also, is your `labels.txt` file in alphabetical order?

**Comment by Sgomes14843** [comment]
Hi, I don't remember well but I think an ACC@1 was between 75 to 90.
The number of images, each class in the train folder has between 700 to 800 images. And my labels are not in alphabetical order, they are like this:
![Anotação 2020-08-05 151006](https://user-images.githubusercontent.com/60934040/89423011-f3ae2d80-d72d-11ea-9df6-a6f09a6c8394.jpg)

**Comment by dusty-nv** [comment]
You need to change the labels to alphabetical order   (the way they would be listed if you did `ls` on your dataset dir)

This is because pytorch doesn't use labels file, rather it loads the class directories in alphanumeric order.

**Comment by Sgomes14843** [comment]
Yes, I already tried it and it worked
Thank you for your help!!

**Comment by dusty-nv** [comment]
OK great, glad that you got it working!
